modules/core/src/main/java/org/apache/synapse/mediators/transform/PayloadFactoryMediator.java

```diff
@@ -132,7 +133,9 @@ private boolean mediate(MessageContext synCtx, String format) {
         }
         org.apache.axis2.context.MessageContext axis2MessageContext = ((Axis2MessageContext) synCtx).getAxis2MessageContext();
         StringBuffer result = new StringBuffer();
-        regexTransform(result, synCtx, format);
+	    StringBuffer resultCTX = new StringBuffer();
```

Comment:
If same variable is used to do the replacing in both (ctx and args) it will append the both results in the payload.